src/capymoa/classifier/_shrubs_ensemble.py

```diff
@@ -0,0 +1,306 @@
+
+        """
+
+        if loss not in ["mse", "ce", "h2"]:
+            raise ValueError(f"Currently only {{mse, ce, h2}} loss is supported, but you provided {loss}")
+        
+        if loss != "mse" and schema.is_regression():
+            raise ValueError(f"For regression tasks only the mse loss is supported, but you provided {loss}.")
+
+        if ensemble_regularizer is not None and ensemble_regularizer not in ["none", "L0", "L1", "hard-L0"]:
+            raise ValueError(f"Currently only {{none, L0, L1, hard-L0}} as ensemble regularizer is supported, but you provided {ensemble_regularizer}.")
+
+        if l_tree_reg < 0:
+            raise ValueError(f"l_tree_reg must be greater or equal to 0, but your provided {l_tree_reg}.")
+
+        if batch_size is None or batch_size < 1:
+            print(f"WARNING: batch_size should be greater than 1, but was {batch_size}. Setting it to 2.")
+            batch_size = 2
+
+        if ensemble_regularizer == "hard-L0" and l_ensemble_reg < 1:
+            print(f"WARNING: You set l_ensemble_reg to `{l_ensemble_reg}', but regularizer is `hard-L0'. In this mode, l_ensemble_reg should be an integer 1 <= l_ensemble_reg <= max_trees where max_trees is the maximum number of estimators you allow in the ensemble")
+
+        if (l_ensemble_reg > 0 and (ensemble_regularizer == "none" or ensemble_regularizer is None)):
+            print(f"WARNING: You set l_ensemble_reg to `{l_ensemble_reg}', but regularizer is None. Ignoring l_ensemble_reg.")
+            l_ensemble_reg = 0
+            
+        if (l_ensemble_reg == 0 and (ensemble_regularizer != "none" and ensemble_regularizer is not None)):
+            print("WARNING: You set l_ensemble_reg to 0, but choose regularizer {}.".format(ensemble_regularizer))
+
+        if isinstance(step_size, str) and step_size != "adaptive":
+            step_size = float(step_size)
+
+
+        # Options set by the user        
+        self.step_size = step_size
+        self.loss = loss
+        self.ensemble_regularizer = ensemble_regularizer
+        self.l_ensemble_reg = l_ensemble_reg
+        self.l_tree_reg = l_tree_reg
+        self.normalize_weights = normalize_weights
+        self.update_leaves = update_leaves
+        self.sk_dt = sk_dt
+        self.batch_size = batch_size
+        self.burnin_steps = burnin_steps
+        self.l_l2_reg = l_l2_reg
+
+        # Number of classes. For regression we simply set this to 1
+        if schema.is_regression():
+            self.n_classes_ = 1
+        else:
+            self.n_classes_ = schema.get_num_classes()
+        
+        self.l_l2_reg = l_l2_reg
+
+        # If the user specifies a random_state, we will also use it. Otherwise we just use 0
+        # TODO This assume that random_state is a number, but it can also be a numpy.randomRandomState. 
```

Comment:
Can we remove this todo?

Reply:
Yes, this TODO can be removed. Initially, I used a dictionary containing the parameters for the scikit-learn model, but then refactored the code to directly use a `{DecisionTreeClassifier, DecisionTreeRegressor}` object combined with `clone`. Note however, that I use the `sk_dt.random_state` in the constructor of the `Classifier` object. I am not sure if this is intended.